pkg/setup/setupmetrics/metrics.go

```diff
@@ -0,0 +1,65 @@
+package setupmetrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/SkycoinProject/skywire-mainnet/pkg/routing"
+)
+
+// Metrics collects metrics for prometheus.
+type Metrics interface {
+	Collectors() []prometheus.Collector
+	RecordRequest(routing.BidirectionalRoute) func(*routing.EdgeRules, *error)
+}
+
+// New returns the default implementation of Metrics.
+func New(namespace string) Metrics {
+	reqCount := prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: namespace,
+		Name:      "active_request_count",
+		Help:      "Current number of ongoing requests.",
+	})
+	reqDurations := prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: namespace,
+		Name:      "request_durations",
+		Help:      "Histogram of request durations.",
+	}, []string{"success"})
+
+	return &metrics{
+		activeRequests: reqCount,
+		reqDurations:   reqDurations,
+	}
+}
+
+type metrics struct {
+	activeRequests prometheus.Gauge
+	reqDurations   prometheus.ObserverVec
+}
+
+func (m *metrics) Collectors() []prometheus.Collector {
+	return []prometheus.Collector{
+		m.activeRequests,
+		m.reqDurations,
+	}
+}
+
+func (m *metrics) RecordRequest(_ routing.BidirectionalRoute) func(rules *routing.EdgeRules, err *error) {
+	start := time.Now()
+	m.activeRequests.Inc()
+
+	return func(rules *routing.EdgeRules, err *error) {
+		var successStr string
```

Comment:
we may simplify this part a bit with something like this:
```
successStr := "true"
if *err != nil {
    successStr = "false"
}
```

btw, is it neccessary to have interface pointer in the func signature, wouldn't just `error` be enough here?

Reply:
That code looks cleaner. I'll get it done.

The pointer of an `error` is needed. We need to get the pointer of the error, as the error value will change.